app/questionnaire/questionnaire_store_updater.py

```diff
@@ -373,53 +385,105 @@ def update_answers(
+        chronological_dependents = self.get_chronological_section_dependents()
+
+        for section in chronological_dependents:
             if (
                 section.section_id,
                 section.list_item_id,
             ) not in self.started_section_keys():
                 continue
 
-            is_path_complete = section.is_complete
-            if is_path_complete is None:
-                is_path_complete = self._router.is_path_complete(
-                    self._router.routing_path(
-                        section.section_id, list_item_id=section.list_item_id
-                    )
+            if (
+                section.section_id,
+                section.list_item_id,
+            ) not in evaluated_dependents:
+                self._evaluate_dependents(
+                    dependent_section=section, evaluated_dependents=evaluated_dependents
                 )
+                evaluated_dependents.append((section.section_id, section.list_item_id))
 
-            self.update_section_status(
-                is_complete=is_path_complete,
-                section_id=section.section_id,
-                list_item_id=section.list_item_id,
+    def _evaluate_dependents(
+        self,
+        *,
+        dependent_section: DependentSection,
+        evaluated_dependents: list[tuple],
+    ) -> None:
+        is_path_complete = dependent_section.is_complete
+        if is_path_complete is None:
+            is_path_complete = self._router.is_path_complete(
+                self._router.routing_path(
+                    dependent_section.section_id,
+                    list_item_id=dependent_section.list_item_id,
+                )
             )
 
+        if self.update_section_status(
+            is_complete=is_path_complete,
+            section_id=dependent_section.section_id,
+            list_item_id=dependent_section.list_item_id,
+        ):
+            dependents_of_dependent = self._schema.when_rules_section_dependencies_by_section_for_progress_value_source[
+                dependent_section.section_id
+            ]
+            for dependent_section_id in dependents_of_dependent:
+                if repeating_list := self._schema.get_repeating_list_for_section(
+                    dependent_section_id
+                ):
+                    for item_id in self._list_store[repeating_list].items:
+                        if (
+                            dependent_section_id,
+                            item_id,
+                        ) not in evaluated_dependents:
+                            self._evaluate_dependents(
+                                dependent_section=DependentSection(
+                                    section_id=dependent_section_id,
+                                    list_item_id=item_id,
+                                    is_complete=None,
+                                ),
+                                evaluated_dependents=evaluated_dependents,
```

Comment:
this is almost identical to the code in the elif so I think you could make a helper method with optional list_item_id or have a list with just the dependent section id and item and then extend it with the repeating list

Reply:
Updated this method 👍